src/utils/solana.ts

```diff
@@ -0,0 +1,31 @@
+import { encodeAddress } from "@polkadot/util-crypto"
+import { PublicKey } from "@solana/web3.js"
+
+export interface SolanaWalletProvider {
+  isPhantom?: boolean
+  isSolflare?: boolean
+  isConnected?: boolean
+  publicKey: PublicKey
+  connect: () => Promise<{ publicKey: PublicKey }>
+  disconnect: () => Promise<void>
+  on: (event: "connect" | "disconnect", handler: (args?: any) => void) => void
+  off: (event: "connect" | "disconnect", handler: (args?: any) => void) => void
+  signTransaction?: (transaction: any) => Promise<any>
+  signAllTransactions?: (transactions: any[]) => Promise<any[]>
+}
+
+export const isPhantom = (provider?: SolanaWalletProvider) => {
+  return !!provider?.isPhantom
+}
+
+export const isSolflare = (provider?: SolanaWalletProvider) => {
+  return !!provider?.isSolflare
+}
+
+export const safeConvertSolanaAddressToSS58 = (address: string, prefix = 0) => {
```

Comment:
What is the reason behind this ? I don't think we can, or need to convert sol address to ss58 ..

Reply:
> What is the reason behind this ? I don't think we can, or need to convert sol address to ss58 ..

So it is compatible with polkadot API calls, when Solana wallet is selected as main account. It is quickest solution for now so we don't get any API errors.